apps/opentelemetry_api/lib/open_telemetry/tracer.ex

```diff
@@ -16,12 +16,39 @@ defmodule OpenTelemetry.Tracer do
       end
   """
 
+  require OpenTelemetry.SemanticConventions.Trace
+  alias OpenTelemetry.SemanticConventions.Trace, as: Conventions
+
+  defmacrop insert_code_attributes(opts) do
+    quote bind_quoted: [opts: opts] do
+      code_function =
+        case __CALLER__.function do
+          {func_name, func_arity} -> "#{func_name}/#{func_arity}"
+          nil -> nil
+        end
+
+      source_attrs = %{
+        Conventions.thread_id() => :erlang.system_info(:scheduler_id),
+        Conventions.code_function() => code_function,
+        Conventions.code_namespace() => __CALLER__.module,
+        Conventions.code_filepath() => __CALLER__.file,
+        Conventions.code_lineno() => __CALLER__.line
+      }
+
+      Map.new()
+      |> Map.update(:attributes, source_attrs, &Map.merge(&1, source_attrs))
```

Comment:
```suggestion
      Map.update(:attributes, source_attrs, source_attrs)
```

Isn't this the same?

Reply:
Oh, I misread entirely :)

Reply:
At least I think... the `&1` applies to the `&Map.merge/2`, right? And no thte pipe?

But I realize I still don't see where the original opt attributes are being merged with.

Reply:
whoopsie, you're absolutely right, I was toying with that while debugging and removed the original opt attributes.